cloud/shared/bin/deploy

```diff
@@ -3,6 +3,10 @@
 set -e
 
 export PROJECT_BASE_DIR="$(realpath "$(dirname "${BASH_SOURCE}")/../../../")"
+
+# Calls validate cloud provider on the things defined in variable_definitions
+# this keeps a list of all the variable definitions
 "${PROJECT_BASE_DIR}/cloud/shared/bin/validate_cloud_provider"
```

Comment:
issue (blocking): at least when this was written, the intention was to *only* validate cloud provider here, as opposed to all of the variables in variable_definitions.json